src/components/Player/index.tsx

```diff
@@ -27,26 +45,83 @@ const Player: FC<{ videoId: string }> = ({ videoId }) => {
 			description: videoTemp.items[0].snippet.description,
 		});
 	}, [videoId]);
+	let clock: IRunningTime | undefined;
+
+	function render({ hours, minutes, seconds }: CountdownRenderProps) {
+		clock = { hours, minutes, seconds };
+		return (
+			<TimerTypography variant="h4">
+				{hours}:{minutes}:{seconds}
+			</TimerTypography>
+		);
+	}
+
+	function updateTime() {
+		if (clock !== undefined) {
+			const hours = convertIntoHours(clock);
+			setTimeAction(hours).then(() => {});
+		}
+	}
+
+	const interval = setInterval(updateTime, 120000);
 
 	useEffect(() => {
-		initialize().then(() => {});
+		initialize().then(() => {
+			getTimerAction().then((data) => {
+				setTime(data);
+			});
+		});
 	}, [initialize]);
+	useEffect(() => {
+		return () => {
+			updateTime();
+			clearInterval(interval);
+		};
+	});
 
 	return (
-		<Box>
+		<>
 			{video.embedHTML ? (
-				<Paper sx={PaperSx}>
-					<YouTube embedHtml={video?.embedHTML} title={video?.title} />
-
-					<Typography variant={"h5"} fontWeight={"600"}>
-						{video?.title}
-					</Typography>
-					<Typography variant={"body1"}>{video.description}</Typography>
-				</Paper>
+				<PlayerContainer>
+					{time?.total_time ? (
+						<PlayerArea>
+							<YouTube embedHtml={video?.embedHTML} title={video?.title} />
+
+							<Typography variant={"h5"} fontWeight={"600"}>
+								{video?.title}
+							</Typography>
+							{video.description.length > 200 ? (
+								<TextArea
+									contentEditable={false}
+									value={
+										video?.description.length > 500 ? video.description.substring(0, 500) + "..." : video.description
+									}
+									rows={10}
+									readOnly
+								/>
+							) : (
+								""
+							)}
+						</PlayerArea>
+					) : (
+						<TimeUpMessage>
+							<SentimentVeryDissatisfiedOutlined />
+							Dear User you have used time quota for today. Please give yourself some time to relax and then come back
+							to watch more videos. <p>{!time?.availed_time ? "You can get 2 more hours... visit settings" : ""}</p>
+						</TimeUpMessage>
+					)}
+					<CountDownBox>
```

Comment:
Create a separate component for clock. Player component should only be responsible for player logic for separation of concerns